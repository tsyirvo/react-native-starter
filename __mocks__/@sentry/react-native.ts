module.exports = {
  ReactNavigationInstrumentation: jest.fn().mockImplementation(() => {}),
};
