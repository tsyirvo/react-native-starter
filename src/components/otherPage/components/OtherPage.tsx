import { useCallback } from 'react';

import { Button, Flex, Text, Title } from '$components/shared/primitives';
import SafeView from '$components/shared/SafeView';
import i18n from '$i18n/config';
import { OtherPageScreenNavigationProp } from '$routes/routes.types';

type Props = {
  navigation: OtherPageScreenNavigationProp;
};

const OtherPage = ({ navigation }: Props) => {
  const goBack = useCallback(() => navigation.goBack(), [navigation]);

  return (
    <SafeView>
      <Flex justifyContent="center" alignItems="center">
        <Title fontWeight={600} testID="otherPage_title" variant="large">
          {i18n.t('otherPage.navigation.title')}
        </Title>

        <Button
          alignItems="center"
          mt="medium"
          testID="back_button"
          onPress={goBack}
        >
          <Text>{i18n.t('otherPage.navigation.backCta')}</Text>
        </Button>
      </Flex>
    </SafeView>
  );
};

export default OtherPage;
