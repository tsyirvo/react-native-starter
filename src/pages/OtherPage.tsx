import { useCallback } from 'react';

import { Box, Button, Text } from '$components/shared/primitives';
import SafeView from '$components/shared/SafeView';
import i18n from '$i18n/config';
import { OtherPageScreenNavigationProp } from '$navigation/navigation.types';

type Props = {
  navigation: OtherPageScreenNavigationProp;
};

const OtherPage = ({ navigation }: Props) => {
  const goBack = useCallback(() => navigation.goBack(), [navigation]);

  return (
    <SafeView>
      <Box justifyContent="center" alignItems="center">
        <Text testID="otherPage_title" variant="large">
          {i18n.t('otherPage.navigation.title')}
        </Text>

        <Box mt="global_24">
          <Button
            alignItems="center"
            testID="back_button"
            onPress={goBack}
            label={i18n.t('otherPage.navigation.backCta')}
          />
        </Box>
      </Box>
    </SafeView>
  );
};

export default OtherPage;
