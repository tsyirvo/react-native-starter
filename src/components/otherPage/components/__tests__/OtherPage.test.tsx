import React from 'react';
import { fireEvent } from '@testing-library/react-native';

import render from '$tests/utils';

import OtherPage from '../OtherPage';

describe('OtherPage page component', () => {
  // given
  const props = { navigation: { goBack: jest.fn() } as any };

  it('should render correctly', () => {
    // When
    const wrapper = render(<OtherPage {...props} />);

    // Then
    expect(wrapper).toMatchSnapshot();
  });

  it('should trigger the goBack method when the button is pressed', () => {
    // When
    const { getByTestId } = render(<OtherPage {...props} />);

    fireEvent.press(getByTestId('back_button'));

    // Then
    expect(props.navigation.goBack).toHaveBeenCalled();
  });
});
