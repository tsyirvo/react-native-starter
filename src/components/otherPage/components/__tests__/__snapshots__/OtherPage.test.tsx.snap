// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OtherPage page component should render correctly 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "display": "flex",
        },
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
          },
        ],
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "display": "flex",
          },
          Array [
            Object {
              "display": "flex",
              "fontFamily": "WorkSans-Bold",
              "fontSize": 24,
              "lineHeight": 24,
            },
          ],
        ]
      }
      testID="otherPage_title"
    >
      Another page
    </Text>
    <View
      style={
        Array [
          Object {
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="back_button"
      >
        <View
          style={
            Array [
              Object {
                "display": "flex",
              },
              Array [
                Object {
                  "display": "flex",
                },
                Array [
                  Object {
                    "alignItems": "center",
                    "marginTop": 24,
                  },
                ],
              ],
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "display": "flex",
                },
                Array [
                  Object {
                    "display": "flex",
                  },
                ],
              ]
            }
          >
            Back
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
